doc/learn/build-a-syntax-tree-typescript.md

```diff
@@ -0,0 +1,130 @@
+---
+group: recipe
+index: 3
+title: Building a content syntax tree
+description: How to build content with syntax trees
+tags:
+  - mdast
+  - hast
+  - xast
+  - builder
+  - hyperscript
+  - jsx
+author: Christian Murphy
+authorGithub: ChristianMurphy
+published: 2020-06-09
+modified: 2020-06-11
+---
+
+## How to build syntax tree
+
+It’s often useful to build new (fragments of) syntax trees when adding or
+replacing content.
+It’s possible to create trees with plain object and array literals (JSON) or
+programmatically with a small utility.
+Finally it’s even possible to use JSX to build trees.
+
+### JSON
+
+The most basic way to create a tree is with plain object and arrays, for some
+extra type safety this can be checked with the types for the given syntax tree
+language, in this case MDAST:
+
+```ts
+import type {Root} from 'mdast'
+
+const mdast: Root = {
```

Comment:
```suggestion
// Note the `: Root` is a TypeScript annotation. Remove it (and the import) for plain JavaScript.
const mdast: Root = {
```

This might be useful, as some beginners might copy/paste it and hit an error?